examples/apps/tree-comparison/src/model/newTreeInventoryList.ts

```diff
@@ -101,8 +101,11 @@ class NewTreeInventoryItem extends TypedEmitter<IInventoryItemEvents> implements
 }
 
 export class NewTreeInventoryList extends DataObject implements IInventoryList {
-	private _sharedTree: ISharedTree | undefined;
-	private get sharedTree(): ISharedTree {
+	on(event: "itemAdded" | "itemDeleted", listener: (item: IInventoryItem) => void): this {
+		throw new Error("Method not implemented.");
+	}
```

Comment:
Not sure why we were getting away with this before, but the TSC error seems legit:
```
Class 'NewTreeInventoryList' incorrectly implements interface 'IInventoryList'.
  Property 'on' is missing in type 'NewTreeInventoryList' but required in type 'IInventoryList'.ts(2420)
modelInterfaces.ts(47, 2): 'on' is declared here.
```
@ChumpChief - FYI.

Reply:
A partial explanation:
* NewTreeInventoryList transitively inherits an implementation of TypedEventEmitter via DataObject.
* However, DataObject has an empty event map by default.
    * Subclasses provide the event map by passing a generic argument (i.e., ... extends DataObject<{ Events: { itemAdded: "itemAdded", itemRemoved: "itemRemoved" }}>
    * Without an event map, the signature is '.on(eventKey: never, ...)'

...so TypeScript was previously only missing the fact that the inheritted '.on(never)' does not satisfy '.on(itemAdded | itemRemoved)'.